core/src/main/java/io/onetable/avro/AvroSchemaConverter.java

```diff
@@ -100,7 +100,7 @@ private OneSchema toOneSchema(
       Schema schema, String parentPath, Map<String, IdMapping> fieldNameToIdMapping) {
     // TODO - Does not handle recursion in Avro schema
     OneType newDataType;
-    Map<OneSchema.MetadataKey, Object> metadata = new HashMap<>();
+    Map<OneSchema.MetadataKey, Object> metadata = new EnumMap<>(OneSchema.MetadataKey.class);
```

Comment:
See my note below about optimization. This code could be updated to take a similar approach to avoid creating a map unless it is required.